node/chainSimulator/components/statusComponents.go

```diff
@@ -87,12 +107,42 @@ func (s *statusComponentsHolder) String() string {
 	return ""
 }
 
-// SetForkDetector will do nothing
-func (s *statusComponentsHolder) SetForkDetector(_ process.ForkDetector) error {
+// SetForkDetector will set the fork detector
+func (s *statusComponentsHolder) SetForkDetector(forkDetector process.ForkDetector) error {
+	if check.IfNil(forkDetector) {
+		return process.ErrNilForkDetector
+	}
+
+	s.forkDetector = forkDetector
+
 	return nil
 }
 
-// StartPolling will do nothing
+// StartPolling starts polling for the updated status
 func (s *statusComponentsHolder) StartPolling() error {
+	var ctx context.Context
```

Comment:
please add a nil check here for the forkdetector or otherwise the code can panic.

Reply:
added